src/Testing/ZopeTestCase/zopedoctest/functional.py

```diff
@@ -237,8 +237,9 @@ def setup_test_class(self):
 
         # If the test_class does not have a runTest method, we add
         # a dummy attribute so that TestCase construction works.
+        # runTest may get called, so set it to something that works.
         if not hasattr(test_class, 'runTest'):
-            setattr(test_class, 'runTest', None)
+            setattr(test_class, 'runTest', lambda x: None)
 
```

Comment:
When I run into this problem, it's usually caused by not defining an explict test suite for the test module. IIRC ZopeTestCase never got support for unit test auto-discovery without test suite declarations. But I never looked into this in detail.